src/m365/cli/commands/cli-doctor.ts

```diff
@@ -21,7 +21,7 @@ interface CliDiagnosticInfo {
   cliVersion: string;
   cliConfig: any;
   roles: string[];
-  scopes: string[];
+  scopes: object
```

Comment:
Hi @dojcsakj, is there a reason you've typed this as an object, instead of keeping it a Map-type?

Reply:
Yes, I have. 

```
    const diagnosticInfo: CliDiagnosticInfo = {
      os: {
        platform: os.platform(),
        version: os.version(),
        release: os.release()
      },
      cliVersion: app.packageJson().version,
      nodeVersion: process.version,
      cliAadAppId: auth.service.appId,
      cliAadAppTenant: validation.isValidGuid(auth.service.tenant) ? 'single' : auth.service.tenant,
      authMode: AuthType[auth.service.authType],
      cliEnvironment: process.env.CLIMICROSOFT365_ENV ? process.env.CLIMICROSOFT365_ENV : '',
      cliConfig: Cli.getInstance().config.all,
      roles: roles,
      scopes: Object.fromEntries(scopes)
    };

    await logger.log(diagnosticInfo);
```

```logger.log()``` uses ```JSON.stringify()``` to created the output but ```JSON.stringify()``` cannot deal with ```Map``` type, so I typed it ```object``` and used ```Object.fromEntries(scopes)``` to pass the scope.

If you have any suggestion that would make ```Map``` type work then please, tell so.

Reply:
No, I kind of checked it myself, and you're correct. It does not work. Let's leave it like this.